### PR TITLE
[KOGITO-4806] - Track which version of the Operator deployed a Kogito…

### PR DESCRIPTION
**Pull request by spolti**
…Runtime
See: https://issues.redhat.com/browse/KOGITO-4806

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ x You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

File: core/deployment/deployment_processor_test.go [hunk 1]
```diff
@@ -75,19 +76,23 @@ func TestNewDeploymentProcessorTesting(t *testing.T) {
...
-	deploymentProcessor := NewDeploymentProcessor(context, runtimeDeployment, RuntimeHandler, Supportingservicehandler)
+	deploymentProcessor := NewDeploymentProcessor(context, runtimeDeployment, runtimeHandler, supportingservicehandler)
 	err := deploymentProcessor.Process()
 
 	assert.NoError(t, err)
 	assert.Equal(t, runtimeDeployment.Spec.Template.Spec.Containers[0].EnvFrom[0].ConfigMapRef.Name, configmap.Name)
 
+	v := make(map[string]string)
+	v[framework.KogitoOperatorVersionAnnotation] = "1.0-SNAPSHOT"
+	assert.Equal(t, v, runtimeDeployment.Annotations)
```
Comment (radtriste) [inline core/deployment/deployment_processor_test.go:95]: could you also check `runtimeDeployment.Template.Annotations` ?

File: core/deployment/deployment_processor.go [hunk 2]
```diff
@@ -69,3 +73,16 @@ func (d *deploymentProcessor) injectSupportingServiceEndpointIntoDeployment() er
...
+
+func (d *deploymentProcessor) injectKogitoAnnotations() error {
+	// annotate the given deployment with the operator version
+	if d.deployment.Annotations == nil {
+		d.deployment.Annotations = make(map[string]string)
+	}
+	if d.deployment.Spec.Template.Annotations == nil {
+		d.deployment.Spec.Template.Annotations = make(map[string]string)
+	}
+	d.deployment.Annotations[framework.KogitoOperatorVersionAnnotation] = d.Context.Version
```
[the remark this answers is not in the document]
Comment (spolti) [reply core/deployment/deployment_processor.go:85]: If it is empty the annotation is not added.

File: core/deployment/deployment_processor_test.go (hunk 1 above)
Comment (spolti) [reply core/deployment/deployment_processor_test.go:95]: done, also added for the supporting services.

File: core/deployment/deployment_processor.go [hunk 3]
```diff
@@ -69,3 +73,20 @@ func (d *deploymentProcessor) injectSupportingServiceEndpointIntoDeployment() er
 	}
 	return nil
 }
+
+func (d *deploymentProcessor) injectKogitoAnnotations() error {
+	if len(d.Context.Version) == 0 {
+		d.Log.Warn("Not able to get detect Kogito version, version annotation will not be set", "Version Label", framework.KogitoOperatorVersionAnnotation)
+	} else {
+		// annotate the given deployment with the operator version
+		if d.deployment.Annotations == nil {
```
Comment (davidesalerno) [inline core/deployment/deployment_processor.go:82]: Why we decided to put the Annotations here and not in:
- Deployment.ObjectMeta.Annotations
- Deployment.Spec.Template.ObjectMeta.Annotations

**Review by vaibhavjainwiz — APPROVED** [review]
lgtm

**Comment by radtriste** [comment]
```
apiVersion: apps/v1
  kind: Deployment
  metadata:
    annotations:
      deployment.kubernetes.io/revision: "2"
      image.openshift.io/triggers: '[{"from":{"kind":"ImageStreamTag","name":"examples-process-quarkus-example-nightly:latest"},"fieldPath":"spec.template.spec.containers[?(@.name==\"process-quarkus-example\")].image"}]'
      kogito-operator.kiegroup.org/version: 2.0.0-snapshot
      kogito.kie.org/runtime: "true"
```
lgtm

**Review by radtriste — APPROVED** [review]

**Pull request merged**